common/tablefunctions.lua

```diff
@@ -256,3 +256,28 @@ if not pairsByKeys then
 		return iter
 	end
 end
+
+if not table.countMem then 
+	--- Estimate how much memory a table (recursively) consumes
+	---@generic V
+	---@param tbl V[]|table<any, V>
+	---@param maxdepth
+	---@return integer number of bytes estimated by this table.
+	function table.countMem(tbl, maxdepth)
+		if maxdepth == 0 then return 0 end
+		maxdepth = maxdepth or 0
+		local ttype = type(tbl)
+		local myMem = 0
+		if ttype == 'table' then 
+			for k,v in pairs(tbl) do 
+				myMem = myMem + table.countMem(k, maxdepth-1) + table.countMem(v, maxdepth-1)
+			end
+			return myMem
+		elseif ttype == 'string' then 
+			return 24 -- cause hash is 8, len is 8, and pointer is 8
```

Comment:
surely more than the hash is kept? e.g. how would concatenation work?

Reply:
Gotta tradeoff with Lua's completely hidden string hashing: 
Duplicate strings are not stored. 
So either count all string lengths too, or count none of them. No way to tell if its a duplicate.

Reply:
Given that this function already counts tables that might be referenced elsewhere (and therefore are stored once despite being used several places, in the same way strings are), I think it would be more consistent to count the memory usage of the content of the string as well.